src/common/utils/usePagination.ts

```diff
@@ -1,21 +1,30 @@
 import { useHistory } from 'react-router-dom';
 import { useCallback } from 'react';
+import { History } from 'history';
 
 const PAGE_SIZE = 10;
 
 export const usePagination = (pageSize = PAGE_SIZE) => {
   const history = useHistory();
-
-  const urlSearchParams = new URLSearchParams(history.location.search);
-  const pageParam = Number(urlSearchParams.get('page') ?? 1);
+  const pageParam = getCurrentPageParam(history);
   const pageIndex = !Number.isNaN(pageParam) ? pageParam - 1 : 0;
   const cursorIndex = pageIndex * pageSize;
   const cursor = cursorIndex > 0 ? btoa(`arrayconnection:${cursorIndex - 1}`) : undefined;
 
   const getPageCount = (connectionsCount: number | null | undefined) =>
     connectionsCount ? Math.ceil(connectionsCount / pageSize) : 1;
 
-  const goToPage = useCallback((pageIndex: number) => history.push({ search: `?page=${pageIndex + 1}` }), [history]);
+  const goToPage = useCallback(
+    (pageIndex: number) => {
+      const newPageNumber = pageIndex + 1;
+      const currentPageNumber = getCurrentPageParam(history);
+
+      if (currentPageNumber !== newPageNumber) {
+        history.push({ search: `?page=${newPageNumber}` });
```

Comment:
Not directly related to the issue, but would it make sense to use `history.replace` here instead? Would make navigating between different views easier, but disable navigating table pages with the browser's history. Alternatively this could be toggled with an optional prop to the hook.

Reply:
Good point. In many web apps back leads you to previous paginated page (push is used), but it may not be the best for all apps, it's hard to say without user feedback :)